lib/ansible/modules/cloud/amazon/lambda_policy.py

```diff
@@ -0,0 +1,494 @@
+
+    qualifier = None
+    if module.params['version'] > 0:
+        qualifier = str(module.params['version'])
+    elif module.params['alias']:
+        qualifier = str(module.params['alias'])
+
+    return qualifier
+
+
+# ---------------------------------------------------------------------------------------------------
+#
+#   Lambda policy permission functions
+#
+# ---------------------------------------------------------------------------------------------------
+
+def get_policy_statement(module, aws):
+    """
+    Checks that policy exists and if so, that statement ID is present or absent.
+
+    :param module:
+    :param aws:
+    :return:
+    """
+
+    client = aws.client('lambda')
+    policy = dict()
+    policy_statement = dict()
+    sid = module.params['statement_id']
+
+    # set API parameters
+    api_params = set_api_params(module, ('function_name', ))
+    qualifier = get_qualifier(module)
+    if qualifier:
+        api_params.update(Qualifier=qualifier)
+
+    # check if function policy exists
+    try:
+        # get_policy returns a JSON string so must convert to dict before reassigning to its key
+        policy_results = client.get_policy(**api_params)
+        policy = json.loads(policy_results.get('Policy', '{}'))
+
+    except (ClientError, ParamValidationError, MissingParametersError) as e:
+        if not e.response['Error']['Code'] == 'ResourceNotFoundException':
+            module.fail_json(msg='Error retrieving function policy: {0}'.format(e))
+
+    if 'Statement' in policy:
+        # Now that we have the policy, check if required permission statement is present and flatten to
+        # simple dictionary if found.
+        for statement in policy['Statement']:
+            if statement['Sid'] == sid:
+                policy_statement['action'] = statement['Action']
+                policy_statement['principal'] = statement['Principal']['Service']
+                try:
+                    policy_statement['source_arn'] = statement['Condition']['ArnLike']['AWS:SourceArn']
+                except KeyError:
+                    pass
+                try:
+                    policy_statement['source_account'] = statement['Condition']['StringEquals']['AWS:SourceAccount']
+                except KeyError:
+                    pass
+                try:
+                    policy_statement['event_source_token'] = statement['Condition']['StringEquals']['lambda:EventSourceToken']
+                except KeyError:
+                    pass
+                break
+
+    return policy_statement
+
+
+def add_policy_permission(module, aws):
+    """
+    Adds a permission statement to the policy.
+
+    :param module:
+    :param aws:
+    :return:
+    """
+
+    client = aws.client('lambda')
+    changed = False
+
+    # set API parameters
+    params = ('function_name', 'statement_id', 'action', 'principal', 'source_arn', 'source_account', 'event_source_token')
+    api_params = set_api_params(module, params)
```

Comment:
I believe this is equivalent to  `api_params = snake_dict_to_camel_dict(module.params)` - `pc` and `set_api_params` should not be required.